packages/flutter_hooks/lib/src/listenable_map.dart

```diff
@@ -0,0 +1,70 @@
+part of 'hooks.dart';
+
+/// Subscribes to a [Listenable]
+/// and execute callback() whenever [Listenable] update.
+/// Subscribe to a Listenable.
+/// When [Listenable] update execute callback(). it result value is return value.
+/// It don't rebuild the widget unless that result of callback() does not changed. so an unnecessary rebuild not occur, performance is better.
+///
+/// ```dart
+/// class Example extends HookWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///    final listenable = useTextEditingController();
+///    final bool textIsEmpty =
+///         useListenableMap(listenable, () => listenable.text.isEmpty);
+///    return Column(
+///       children: [
+///         TextField(controller: listenable),
+///         ElevatedButton(
+///             // if textIsEmpty is false,　the button is disabled.
+///             onPressed: textIsEmpty ? null : () => print("Pressed!"),
+///             child: Text("Button")),
+///       ],
+///     );
+///   }
+/// }
+/// ```
+R useListenableMap<T extends Listenable, U extends R Function(), R>(
```

Comment:
That's too many generics.

Only R should be necessary, the others shouldn't be here.

Reply:
> Thanks, awesome work 😄
> 
> I don't like the name though. It's not really a "map". Maybe useListenableSelector?

Thank you, Remi 😄
I'm so glad to hear you say that!
I prefer useListenableSelector as naming. I am rethinking the naming.

Thank you for your many suggestions for my code, even though you seem to be busy. I will fix it.